components/custom_status_modal/index.js

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+/* eslint-disable camelcase */
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import {setStatus} from 'mattermost-redux/actions/users';
+import {getStatusForUserId} from 'mattermost-redux/selectors/entities/users';
+
+import {closeModal} from 'actions/views/modals';
+
+import CustomStatusModal from './custom_status_modal.jsx';
+
+function mapStateToProps(state) {
+    const {entities: {users: {currentUserId, customStatuses}}, views: {modals: {modalState: {custom_status}}}} = state;
```

Comment:
Does customStatuses need to be its top level thing in `users`? Can it be part of `statuses`?